src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -40,66 +40,98 @@
 
 @WebServlet("/login")
 public class AuthServlet extends HttpServlet {
-  
 
   /**
-  * doGet checks if the user is currently logged in, and if the user has already a profile. It then returns the correct header
-  */
+   * doGet checks if the user is currently logged in, and if the user has already a profile. It then returns the correct header
+   */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws ServletException, IOException {
     UserService userService = UserServiceFactory.getUserService();
     HttpSession session = request.getSession();
 
     if (userService.isUserLoggedIn()) {
-       handleLogIn(response, session, userService);
+      handleLogIn(response, session, userService);
     } else {
-       session.invalidate();
-       response.sendRedirect("/");
+      session.invalidate();
+      response.sendRedirect("/");
     }
   }
 
-  public void handleLogIn(HttpServletResponse response, HttpSession session, UserService userService) throws IOException {
-       String url = "/login";
+  public void handleLogIn(
+    HttpServletResponse response,
+    HttpSession session,
+    UserService userService
+  )
+    throws IOException {
+    String urlToRedirectToAfterUserLogsOut = "/login";
 
-       Entity currentUser = queryForUser(userService);
-       if (Objects.isNull(currentUser)){
-             setLogInAttributes(0, session, url, userService);
-             response.sendRedirect("/profile_creation.jsp");
-        } else {
-            setLogInAttributes(1, session, url, userService);
-            setUserAttributes(currentUser, session);
-            response.sendRedirect("/user?id="+currentUser.getKey().getId());                
-        }
+    Entity currentUser = queryForUser(userService);
+    if (Objects.isNull(currentUser)) {
+      setRegistartionAttributes(
+        userService,
+        urlToRedirectToAfterUserLogsOut,
+        session
+      );
+      //TODO redirect response to profile creation . js
```

Comment:
If the redirection is already implemented (as I see in the line 77), you should remove this comment.